File: briareos-master/dev/project/briareos/zworker.py
from briareos.core import distributed_zmq
from briareos.core import processing

from briareos.utils import logger
from briareos.utils import loader
from briareos.common import *

import psutil
import uuid


class ZWorker:
    def __init__(self, worker_config_path=ZWORKER_CONFIG_PATH):
        logger.info("Initializing Briareos Z-Worker")
        self.id = uuid.uuid1()
        self.config = loader.import_zworker_config(worker_config_path)
        self.processing_engine = processing.WorkerEngine(self)
        self.workers = []
        self._init_workers()

    def _init_workers(self):
        max_workers = self.get_max_workers()

        for i in range(max_workers):
            self.workers.append(distributed_zmq.Worker(self, self.config))

    def start(self):
        self.processing_engine.start()

        workers_info = "subworkers"
        max_workers = len(self.workers)
        if max_workers == 1:
            workers_info = "subworker"

        logger.info("Starting %s %s" % (max_workers, workers_info))
        for worker in self.workers:
            worker.start()
            logger.subinfo("Subworker ID: %s" % worker.id)

        logger.info("Z-Worker ID: %s" % self.id)
        logger.success("Briareos Z-Worker is running")

    def stop(self):
        logger.info("Stopping Z-Worker...")
        logger.subinfo("Stopping %s workers..." % len(self.workers))
        for worker in self.workers:
            worker.stop()
        self.processing_engine.stop()

    def process(self, packet_payload, pipeline_name):
        return self.processing_engine.process(packet_payload, pipeline_name)

    @staticmethod
    def get_max_workers():
        return 1
        return psutil.cpu_count()

File: jbriareos-master/Tadeu_Freitas-jbriareos-c16b99c2c2d9/src/main/java/components/ZWorker.java
package components;

import core.DistributedZMQ;

import java.util.LinkedList;
import java.util.UUID;

public class ZWorker {
    public UUID id;
    public LinkedList<DistributedZMQ.Worker> workers;

    public ZWorker() {
        System.out.println("Initializing Briareos ZWorker");
        this.id = UUID.randomUUID();
        this.workers = new LinkedList<>();
        this.initWorkers();
    }

    private int getMaxWorkers() {
        return 1;
        // return Runtime.getRuntime().availableProcessors();
    }

    private void initWorkers() {
        int maxWorkers = this.getMaxWorkers();

        for (int i = 0; i < maxWorkers; i++) {
            this.workers.addLast(new DistributedZMQ.Worker(this.id));
        }
    }

    public void start() {
        for (DistributedZMQ.Worker worker : this.workers)
            worker.start();

        System.out.println("ZWorker ID: " + this.id);
        System.out.println("Briareos ZWorker is running");
    }

    public void stop() {
        for (DistributedZMQ.Worker worker : workers)
            worker.stop();
    }
}
